### PR TITLE
add 7zip

### PR DESCRIPTION
**Pull request by developer-guy**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [x] REQUIRED - The package is added to `packages.txt`

#### For security-related PRs

- [ ] The security fix is recorded in `advisories` and `secfixes`

#### For version bump PRs

- [ ] The `epoch` field is reset to 0
- [ ] Patch source: _patch source here_

### CODE REVIEW

File: 7zip.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      make -f ../../cmpl_gcc.mak \
+      	CC="${CC:-cc} $CFLAGS $LDFLAGS" \
+      	CXX="${CXX:-c++} $CXXFLAGS $LDFLAGS" \
+      	DISABLE_RAR=1
+
+  - runs: |
+      install -Dm755 CPP/7zip/Bundles/Alone2/b/g/7zz "${{targets.destdir}}"/usr/bin/7zz
+      ln -s "${{targets.destdir}}"/usr/bin/7zz "${{targets.destdir}}"/usr/bin/7z
+      ls -latr
+      install -Dm644 DOC/* -t "${{targets.destdir}}"/usr/share/doc/7zip
```
Comment (amouat) [inline 7zip.yaml:45]: should this be here? It's in the doc bit later, right?

File: 7zip.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      expected-sha512: 3f391b1bd65a0654eb5b31b50f1d400f0ec38ab191d88e15849a6e4d164b7bf2ce4a6d70ec8b6e27bde1b83bb2d45b65c03129499334669e05ee025784be455a
+      strip-components: 0
+
+  - name: Configure and build
+    runs: |
+      cd CPP/7zip/Bundles/Alone2
+      mkdir -p b/g
+      # TODO: enable asm (requires jwasm or uasm)
+      # DISABLE_RAR: RAR codec is non-free
+      # -D_GNU_SOURCE: broken sched.h defines
```
Comment (amouat) [inline 7zip.yaml:35]: I'd delete this line to avoid confusing people

File: 7zip.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    with:
+      uri: https://7-zip.org/a/7z${{vars.mangled-package-version}}-src.tar.xz
+      expected-sha512: 3f391b1bd65a0654eb5b31b50f1d400f0ec38ab191d88e15849a6e4d164b7bf2ce4a6d70ec8b6e27bde1b83bb2d45b65c03129499334669e05ee025784be455a
+      strip-components: 0
+
+  - name: Configure and build
+    runs: |
+      cd CPP/7zip/Bundles/Alone2
+      mkdir -p b/g
+      # TODO: enable asm (requires jwasm or uasm)
```
Comment (amouat) [inline 7zip.yaml:33]: I'm curious about this. Do we know what it would require and how hard it would be? (not a blocker)

File: 7zip.yaml (hunk 1 above)
Comment (developer-guy) [reply 7zip.yaml:45]: I'm not sure but in the APKBUILD file it says it provides "pkgname"-doc, so, I added here as well.

**Review by kaniini — APPROVED** [review]

**Review by kaniini — APPROVED** [review]

**Pull request merged**